manifests/cli.pp

```diff
@@ -0,0 +1,63 @@
+# = Pulpcore command line interface
+#
+# This class installs the Pulpcore command line interface (CLI).
+#
+# === Parameters:
+#
+# $pulpcore_url::       URL on which Pulpcore runs
+#
+# $username::           Username for authentication
+#
+# $password::           Password for authentication
+#
+# $cert::               Client certificate for authentication
+#
+# $key::                Client key for authentication
+#
+# === Advanced parameters:
+#
+# $api_root::           Absolute API base path on server (not including 'api/v3/')
+#
+# $verify_ssl::         Verify SSL connection
+#
+# $dry_run::            Trace commands without performing any unsafe HTTP calls
+#
+# $version::            pulp-cli package version, it's passed to ensure parameter of package resource
+#                       can be set to specific version number, 'latest', 'present' etc.
+#
+class pulpcore::cli (
+  Optional[Stdlib::HTTPUrl] $pulpcore_url = undef,
+  String $version = 'installed',
+  Optional[String] $username = undef,
+  Optional[String] $password = undef,
+  Optional[String] $cert = undef,
+  Optional[String] $key = undef,
+  Optional[String] $api_root = undef,
+  Boolean $verify_ssl = true,
+  Boolean $dry_run = true,
+) {
+  package { 'pulp-cli':
+    ensure => $version,
+  }
+  -> file { '/etc/pulp/cli.toml':
```

Comment:
Do we also want to manage `/root/.config/pulp/cli.toml` like we do for Hammer? Given the default to cert auth, I'd vote no?

Reply:
The certs are `root:foreman`, so certainly usable by others than `root`, thus it felt "OK" to put them in the global config?

Reply:
Turns out, if you specify both certs (in global config) and password (in user config) things explode. So I think I'll move the certs to the user config after all.